src/queries/text.ts

```diff
@@ -58,37 +59,65 @@ const getChildrenAsText = (
 const getNodeByText = (
   node: ReactTestInstance,
   text: TextMatch,
+  TextComponent: React.ComponentType,
   options: TextMatchOptions = {}
 ) => {
-  try {
-    const { Text } = require('react-native');
-    const isTextComponent = filterNodeByType(node, Text);
-    if (isTextComponent) {
-      const textChildren = getChildrenAsText(node.props.children, Text);
-      if (textChildren) {
-        const textToTest = textChildren.join('');
-        const { exact, normalizer } = options;
-        return matches(text, textToTest, normalizer, exact);
-      }
+  const isTextComponent = filterNodeByType(node, TextComponent);
+  if (isTextComponent) {
+    const textChildren = getChildrenAsText(node.props.children, TextComponent);
+    if (textChildren) {
+      const textToTest = textChildren.join('');
+      const { exact, normalizer } = options;
+      return matches(text, textToTest, normalizer, exact);
     }
-    return false;
-  } catch (error) {
-    throw createLibraryNotSupportedError(error);
   }
+  return false;
 };
 
+function getCompositeParent(
+  element: ReactTestInstance,
+  compositeType: React.ComponentType
+) {
+  if (!isHostElement(element)) return null;
```

Comment:
IMO we can get rid of this check, as it's not needed in the code and also it's hard to guess such check exists based on function name.
```suggestion
  if (!isHostElement(element)) return null;
```